File: lingting-security-grpc/src/main/java/live/lingting/framework/properties/SecurityGrpcProperties.java
package live.lingting.framework.properties;

import io.grpc.Metadata;
import lombok.Getter;
import lombok.Setter;

/**
 * @author lingting 2023-12-14 16:40
 */
@Getter
@Setter
public class SecurityGrpcProperties {

	private String authorizationKey = "Authorization";

	public Metadata.Key<String> authorizationKey() {
		return Metadata.Key.of(getAuthorizationKey(), Metadata.ASCII_STRING_MARSHALLER);
	}

}

File: lingting-security-grpc/src/main/kotlin/live/lingting/framework/security/grpc/properties/SecurityGrpcProperties.kt
package live.lingting.framework.security.grpc.properties

import io.grpc.Metadata

/**
 * @author lingting 2023-12-14 16:40
 */
class SecurityGrpcProperties {
    var authorizationKey: String = "Authorization"

    fun authorizationKey(): Metadata.Key<String> {
        return Metadata.Key.of(authorizationKey, Metadata.ASCII_STRING_MARSHALLER)
    }
}
